models/review.go

```diff
@@ -218,17 +239,46 @@ func isOfficialReviewer(e Engine, issue *Issue, reviewer *User) (bool, error) {
 	return pr.ProtectedBranch.isUserOfficialReviewer(e, reviewer)
 }
 
+// IsOfficialReviewerTeam check if reviewer in this team can make official reviews in issue (counts towards required approvals)
+func IsOfficialReviewerTeam(issue *Issue, team *Team) (bool, error) {
+	return isOfficialReviewerTeam(x, issue, team)
+}
+
+func isOfficialReviewerTeam(e Engine, issue *Issue, team *Team) (bool, error) {
+	pr, err := getPullRequestByIssueID(e, issue.ID)
+	if err != nil {
+		return false, err
+	}
+	if err = pr.loadProtectedBranch(e); err != nil {
+		return false, err
+	}
+	if pr.ProtectedBranch == nil {
+		return false, nil
+	}
+
+	if !pr.ProtectedBranch.EnableApprovalsWhitelist {
+		return team.Authorize >= AccessModeWrite, nil
+	}
+
+	return base.Int64sContains(pr.ProtectedBranch.ApprovalsWhitelistTeamIDs, team.ID), nil
+}
+
 func createReview(e Engine, opts CreateReviewOptions) (*Review, error) {
 	review := &Review{
-		Type:       opts.Type,
-		Issue:      opts.Issue,
-		IssueID:    opts.Issue.ID,
-		Reviewer:   opts.Reviewer,
-		ReviewerID: opts.Reviewer.ID,
-		Content:    opts.Content,
-		Official:   opts.Official,
-		CommitID:   opts.CommitID,
-		Stale:      opts.Stale,
+		Type:         opts.Type,
+		Issue:        opts.Issue,
+		IssueID:      opts.Issue.ID,
+		Reviewer:     opts.Reviewer,
+		ReviewerTeam: opts.ReviewerTeam,
+		Content:      opts.Content,
+		Official:     opts.Official,
+		CommitID:     opts.CommitID,
+		Stale:        opts.Stale,
+	}
+	if opts.Reviewer != nil {
+		review.ReviewerID = opts.Reviewer.ID
+	} else {
+		review.ReviewerTeamID = opts.ReviewerTeam.ID
```

Comment:
It also seems to me that the only valid team request is a ReviewTypeRequest - should we be enforcing that?

Reply:
> Do we need to check that the reviewerTeam is actually a team of this organization?

Maybe not needed because the team which is not in this org won't have access permisson to the repos in this org. which won't pass previous permission check.

Reply:
OK. How about the ReviewTypeRequest question?

Reply:
Hmm, I think so